frontend/src/components/eMIB/EditActionDialog.jsx

```diff
@@ -119,19 +124,127 @@ class EditActionDialog extends Component {
     this.setState({ action: updatedAction });
   };
 
-  showCancelConfirmationDialog = () => {
-    this.setState({ showCancelConfirmationDialog: true });
+  /* 
+  this handles the display of the cancel confirmation message: Are you sure you want to cancel this response?
+  when creating a new email/task and trying to exit the form, if it's empty, don't display any warning message
+  when creating a new email/task and trying to exit the form, if there is any change to at lest one of the fields, display the cancel warning message
+  when editing an email/task and trying to exit the form, if there is no changes, don't display any warning messsage
+  when editing an email/task and trying to exit the form, if there is any change to at lest one of the fields, display the cancel warning message
+  */
+  handleCancelConfirmationDisplay = () => {
```

Comment:
Wondering how hard it would be to export and test this function... It'd be great to have tests specifically for this function, but it would not be easy to pull off (seeing as you need to get props, etc.).

Reply:
As discussed offline, we'll pair tomorrow morning to work on these new tests.